lib/PageInteractions.pm

```diff
@@ -0,0 +1,55 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: TBD
+# Maintainer: Oleksandr Orlov <oorlov@suse.de>
+
+package PageInteractions;
+use strict;
+use warnings FATAL => 'all';
+
+use testapi;
+use parent 'Exporter';
+use Exporter;
+
+our @EXPORT = qw(
+    find
+    send_keys
+    fill_in_password
+    is_displayed
+);
+
+sub find {
+    my $self = @_;
+    assert_screen(@_);
+}
+
+sub send_keys {
+    my $key = shift;
+    my $message = shift || "Send $key key";
+    record_info('send_key', $message);
+    send_key($key, @_);
+}
+
+sub fill_in_password {
+    my $password = shift;
+    my $message = shift || "Type password";
+    my $self = @_;
```

Comment:
We follow `my ($self) = @_;` style everywhere else, don't you mind to change it? We don't need `$self` here either.

Reply:
Again my poor Perl skills, I wanted to pass the same arguments to type_password method and add some specific parameters (as $message) to the extended _fill_in_password_ and didn't find quick solution how to do this. Probably @_[1]

Reply:
@OleksandrOrlov Take a look in https://github.com/os-autoinst/os-autoinst/tree/master/testapi.pm . I am sure you will find good examples for parameter parsing including the according documentation and "support for legacy behaviour" in there.